src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Dependency.cs

```diff
@@ -266,7 +266,9 @@ private static string GetAlias(IDependency dependency)
 
         private static string Normalize(string id)
         {
-            return id.Replace('/', '\\');
+            return id
+                .Replace('/', '\\')
+                .Replace("..", "__");
```

Comment:
Okay, I thought about this more, I can see why analyzers did this originally, analyzers look like the following in the project file:

``` C#
<ItemGroup>
    <Analyzer Include="Analyzer..dll"/>
</ItemGroup>
```

This looks a lot like like source files:

``` XML
<ItemGroup>
    <Compile Include="Foo.cs"/>
</ItemGroup>
```

I can see why we return the analyzer's canonical name as the same as the path of the analyzer. However, I will state - we should not be expecting or interpreting that outside of the project system. There are other APIs for retrieving a file path given a canonical/mkdocument name, we should be using that. We're theoretically allowed to change return whatever we want for a canonical name, for example, that canonical name would be problematic in the following:

``` C#
<ItemGroup>
    <Analyzer Include="Analyzer..dll"/>
    <None Include="Analyzer.dll" /> 
</ItemGroup>
```

These are going to have the same canonical name even though they are supposed to be unique.